// graphql-to-openapi/dist/tests/basics/customScalars.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var fs_1 = require("fs");
var path = require("path");
var index_1 = require("../../index");
var assert = require("assert");
var stringify = require("json-stable-stringify");
describe('customScalars', function () {
    var schema = (0, fs_1.readFileSync)(path.join(__dirname, 'customScalarsSchema.graphql')).toString();
    var inputQueryFilename = path.join(__dirname, 'customScalars.graphql');
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    var expectedOutput = require('./customScalars.json');
    var query = (0, fs_1.readFileSync)(inputQueryFilename).toString();
    var scalarConfig = {
        CustomScalar: {
            type: 'string',
        },
    };
    var normalizedExpectedOutput = stringify(expectedOutput, { space: '  ' });
    it('should produce a valid openapi spec with a supplied scalarConfig', function () {
        var actualOutput = (0, index_1.graphqlToOpenApi)({
            schema: schema,
            query: query,
            scalarConfig: scalarConfig,
        }).openApiSchema;
        var normalizedActualOutput = stringify(actualOutput, { space: '  ' });
        assert.ok(!!actualOutput);
        assert.equal(normalizedActualOutput, normalizedExpectedOutput);
    });
    it('should produce a valid openapi spec with an onUnknownScalar function', function () {
        function onUnknownScalar(s) {
            return scalarConfig[s];
        }
        var actualOutput = (0, index_1.graphqlToOpenApi)({
            schema: schema,
            query: query,
            onUnknownScalar: onUnknownScalar,
        }).openApiSchema;
        var normalizedActualOutput = stringify(actualOutput, { space: '  ' });
        assert.ok(!!actualOutput);
        assert.equal(normalizedActualOutput, normalizedExpectedOutput);
    });
    it('should throw an exception when a custom scalar is unhandled', function () {
        try {
            (0, index_1.graphqlToOpenApi)({
                schema: schema,
                query: query,
            }).openApiSchema;
            assert.ok(false);
        }
        catch (err) {
            assert.ok(err);
        }
    });
});
